### PR TITLE
Add makefile in /src

### PR DESCRIPTION
**Pull request by cw9**
Move golang related build targets to /src/Makefile

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
+# To test echo parser:
 # make test-impl cmd-parser-text=echo
 #
 # To test github.com/prometheus/client_python parser:
 # make prometheus_client_python_parser test-impl cmd-parser-text="docker run --rm -i prometheus_client_python_parser:latest"
 .PHONY: test-impl
-test-impl: openmetricstest
+test-impl:
+# openmetricstest needs to be built in /src since it requires /src/go.mod
+	(cd ./src && make openmetricstest)
```
Comment (robskillington) [inline Makefile:13]: nit: Think each line in the makefile executes as a subshell already so can do this without the brackets which is slightly cleaner. i.e. `cd ./src && make openmetricstest`

File: specification/OpenMetrics.txt [hunk 2]
```diff
@@ -5,12 +5,12 @@
 Ops WG                                                  R. Hartmann, Ed.
 Internet-Draft                                              Grafana Labs
 Intended status: Standards Track                               B. Kochie
-Expires: June 20, 2021                                            GitLab
+Expires: December 10, 2021                                        GitLab
```
Comment (robskillington) [inline specification/OpenMetrics.txt:8]: Hm interesting, this date looks wrong. Maybe revert this file completely? It must have been regenerated during the move but I don't think we should regen it. This current version has been submitted for IETF for review.

File: src/go.mod [hunk 3]
```diff
@@ -0,0 +1,3 @@
+module github.com/cw9/OpenMetrics/src
```
Comment (robskillington) [inline src/go.mod:1]: This should probably be `github.com/OpenObservability/OpenMetrics/src` before we merge I think?

If this is causing issues locally for you not matching your fork you can change the directory locally the repo is in and just add `github.com/cw9/OpenMetrics` as a git remote. Let me know if facing any issues with that.

**Comment by robskillington** [comment]
Just minor comments, looks good 🙌

File: specification/OpenMetrics.txt (hunk 2 above)
Comment (cw9) [reply specification/OpenMetrics.txt:8]: I have to regen it to pass the test

Comment (cw9) [reply specification/OpenMetrics.txt:8]: It was failing like this: https://app.circleci.com/pipelines/github/OpenObservability/OpenMetrics/119/workflows/b71981b0-1648-45e2-8e43-ecb5b57393d2/jobs/155

File: Makefile (hunk 1 above)
Comment (cw9) [reply Makefile]: done

File: src/go.mod (hunk 3 above)
Comment (cw9) [reply src/go.mod]: done

**Review by robskillington — APPROVED** [review]
LGTM

**Pull request merged**